subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/modulecache/DefaultCachedMetaData.java

```diff
@@ -27,9 +27,9 @@
     private final ModuleSource moduleSource;
     private final BigInteger descriptorHash;
     private final long ageMillis;
-    private final ModuleComponentResolveMetadata metaData;
+    private final MutableModuleComponentResolveMetadata metaData;
 
-    public DefaultCachedMetaData(ModuleMetadataCacheEntry entry, ModuleComponentResolveMetadata metaData, BuildCommencedTimeProvider timeProvider) {
+    public DefaultCachedMetaData(ModuleMetadataCacheEntry entry, MutableModuleComponentResolveMetadata metaData, BuildCommencedTimeProvider timeProvider) {
```

Comment:
`DefaultCachedMetaData` appears to be only used to entries coming _off_ the cache? It's `ModuleMetadataCacheEntry` that's used, isn't it?